osbs/utils.py

```diff
@@ -289,6 +289,10 @@ def get_time_from_rfc3339(rfc3339):
         return timegm(time_tuple)
 
 
+
+VALID_BUILD_CONFIG_NAME_CHARS = re.compile('[a-z0-9]')
```

Comment:
The benefit of not just allocating half the remaining length to each of the two components, and instead adding characters from each one by one until the length limit is reached, is that you get a more readable name in the case that one component of the name is over half the remaining length.